ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt

```diff
@@ -259,6 +329,28 @@ data class PathSegmentOptionalParameterRouteSelector(val name: String, val prefi
     }
 
     override fun toString(): String = "${prefix ?: ""}{$name?}${suffix ?: ""}"
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as PathSegmentOptionalParameterRouteSelector
+
+        if (name != other.name) return false
+        if (prefix != other.prefix) return false
+        if (suffix != other.suffix) return false
+        if (hasTrailingSlash != other.hasTrailingSlash) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = name.hashCode()
```

Comment:
consider extracting `combine` for hashcodes

Reply:
this code is autogenerated by IDE and I would rather avoid making changes there. As discussed offline, I added comment about this.